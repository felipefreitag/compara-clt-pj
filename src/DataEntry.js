import React from 'react';
import { Table, TableBody, TableHeader, TableFooter } from 'material-ui/Table';
import { TableHeaderColumn, TableRow, TableRowColumn } from 'material-ui/Table';
import TextInput from './TextInput';
import RaisedButton from 'material-ui/RaisedButton';

const DataEntry = props => {
  const { showResult } = props
  return (
    <Table id='inputs' selectable={false}>
      <TableHeader
        displaySelectAll={false}
        adjustForCheckbox={false}
      >
        <TableRow>
          <TableHeaderColumn style={{textAlign:'center'}}>
            CLT
          </TableHeaderColumn>
          <TableHeaderColumn style={{textAlign:'center'}}>
            PJ
          </TableHeaderColumn>
        </TableRow>
      </TableHeader>
      <TableBody displayRowCheckbox={false}>
        <TableRow>
          <TableRowColumn>
            <TextInput onChange={props.handleEmployeeChange} />
          </TableRowColumn>
          <TableRowColumn>
            <TextInput onChange={props.handleCompanyChange} />
          </TableRowColumn>
        </TableRow>
      </TableBody>
      <TableFooter adjustForCheckbox={false}>
        <TableRow style={{borderTop: '0px'}}>
          <TableRowColumn style={{ textAlign: 'center' }}>
              <RaisedButton
                label={showResult ?
                  "Ocultar resultados" : "Calcular"}
                primary={true}
                onClick={props.calculate}
              />
          </TableRowColumn>
        </TableRow>
      </TableFooter>
    </Table>

  );
}
export default DataEntry
