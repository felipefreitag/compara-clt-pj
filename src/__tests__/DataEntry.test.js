import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DataEntry from '../DataEntry'

Enzyme.configure({ adapter: new Adapter() })

it('renders correctly', () => {
  const wrapper = shallow(<DataEntry />)
  expect(wrapper).toMatchSnapshot()
})
