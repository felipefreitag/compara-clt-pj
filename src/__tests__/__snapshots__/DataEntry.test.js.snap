// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataEntry />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "allRowsSelected": false,
      "children": Array [
        <TableHeader
          adjustForCheckbox={false}
          displaySelectAll={false}
          enableSelectAll={true}
          selectAllSelected={false}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableHeaderColumn
                              style={
                                        Object {
                                                  "textAlign": "center",
                                                }
                              }
                    >
                              CLT
                    </TableHeaderColumn>
                    <TableHeaderColumn
                              style={
                                        Object {
                                                  "textAlign": "center",
                                                }
                              }
                    >
                              PJ
                    </TableHeaderColumn>
          </TableRow>
</TableHeader>,
        <TableBody
          allRowsSelected={false}
          deselectOnClickaway={true}
          displayRowCheckbox={false}
          multiSelectable={false}
          preScanRows={true}
          selectable={true}
          style={Object {}}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
          >
                    <TableRowColumn
                              hoverable={false}
                    >
                              <TextInput
                                        label="Sal치rio bruto CLT"
                                        onChange={undefined}
                              />
                    </TableRowColumn>
                    <TableRowColumn
                              hoverable={false}
                    >
                              <TextInput
                                        label="Faturamento mensal MEI"
                                        onChange={undefined}
                              />
                    </TableRowColumn>
          </TableRow>
</TableBody>,
        <TableFooter
          adjustForCheckbox={false}
          style={Object {}}
>
          <TableRow
                    displayBorder={true}
                    hoverable={false}
                    hovered={false}
                    selectable={true}
                    striped={false}
                    style={
                              Object {
                                        "borderTop": "0px",
                                      }
                    }
          >
                    <TableRowColumn
                              hoverable={false}
                              style={
                                        Object {
                                                  "textAlign": "center",
                                                }
                              }
                    >
                              <RaisedButton
                                        disabled={false}
                                        fullWidth={false}
                                        label="Calcular"
                                        labelPosition="after"
                                        onClick={undefined}
                                        primary={true}
                                        secondary={false}
                              />
                    </TableRowColumn>
          </TableRow>
</TableFooter>,
      ],
      "fixedFooter": true,
      "fixedHeader": true,
      "height": "inherit",
      "id": "inputs",
      "multiSelectable": false,
      "selectable": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "adjustForCheckbox": false,
          "children": <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableHeaderColumn
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            >
                        CLT
            </TableHeaderColumn>
            <TableHeaderColumn
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            >
                        PJ
            </TableHeaderColumn>
</TableRow>,
          "displaySelectAll": false,
          "enableSelectAll": true,
          "selectAllSelected": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableHeaderColumn
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
>
                CLT
</TableHeaderColumn>,
              <TableHeaderColumn
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
>
                PJ
</TableHeaderColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "CLT",
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "CLT",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "PJ",
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "PJ",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "allRowsSelected": false,
          "children": <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
>
            <TableRowColumn
                        hoverable={false}
            >
                        <TextInput
                                    label="Sal치rio bruto CLT"
                                    onChange={undefined}
                        />
            </TableRowColumn>
            <TableRowColumn
                        hoverable={false}
            >
                        <TextInput
                                    label="Faturamento mensal MEI"
                                    onChange={undefined}
                        />
            </TableRowColumn>
</TableRow>,
          "deselectOnClickaway": true,
          "displayRowCheckbox": false,
          "multiSelectable": false,
          "preScanRows": true,
          "selectable": true,
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TableRowColumn
                hoverable={false}
>
                <TextInput
                                label="Sal치rio bruto CLT"
                                onChange={undefined}
                />
</TableRowColumn>,
              <TableRowColumn
                hoverable={false}
>
                <TextInput
                                label="Faturamento mensal MEI"
                                onChange={undefined}
                />
</TableRowColumn>,
            ],
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <TextInput
                  label="Sal치rio bruto CLT"
                  onChange={undefined}
/>,
                "hoverable": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "label": "Sal치rio bruto CLT",
                  "onChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <TextInput
                  label="Faturamento mensal MEI"
                  onChange={undefined}
/>,
                "hoverable": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "label": "Faturamento mensal MEI",
                  "onChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "adjustForCheckbox": false,
          "children": <TableRow
            displayBorder={true}
            hoverable={false}
            hovered={false}
            selectable={true}
            striped={false}
            style={
                        Object {
                                    "borderTop": "0px",
                                  }
            }
>
            <TableRowColumn
                        hoverable={false}
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            >
                        <RaisedButton
                                    disabled={false}
                                    fullWidth={false}
                                    label="Calcular"
                                    labelPosition="after"
                                    onClick={undefined}
                                    primary={true}
                                    secondary={false}
                        />
            </TableRowColumn>
</TableRow>,
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <TableRowColumn
              hoverable={false}
              style={
                            Object {
                                          "textAlign": "center",
                                        }
              }
>
              <RaisedButton
                            disabled={false}
                            fullWidth={false}
                            label="Calcular"
                            labelPosition="after"
                            onClick={undefined}
                            primary={true}
                            secondary={false}
              />
</TableRowColumn>,
            "displayBorder": true,
            "hoverable": false,
            "hovered": false,
            "selectable": true,
            "striped": false,
            "style": Object {
              "borderTop": "0px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Calcular"
                labelPosition="after"
                onClick={undefined}
                primary={true}
                secondary={false}
/>,
              "hoverable": false,
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Calcular",
                "labelPosition": "after",
                "onClick": undefined,
                "primary": true,
                "secondary": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "allRowsSelected": false,
        "children": Array [
          <TableHeader
            adjustForCheckbox={false}
            displaySelectAll={false}
            enableSelectAll={true}
            selectAllSelected={false}
>
            <TableRow
                        displayBorder={true}
                        hoverable={false}
                        hovered={false}
                        selectable={true}
                        striped={false}
            >
                        <TableHeaderColumn
                                    style={
                                                Object {
                                                            "textAlign": "center",
                                                          }
                                    }
                        >
                                    CLT
                        </TableHeaderColumn>
                        <TableHeaderColumn
                                    style={
                                                Object {
                                                            "textAlign": "center",
                                                          }
                                    }
                        >
                                    PJ
                        </TableHeaderColumn>
            </TableRow>
</TableHeader>,
          <TableBody
            allRowsSelected={false}
            deselectOnClickaway={true}
            displayRowCheckbox={false}
            multiSelectable={false}
            preScanRows={true}
            selectable={true}
            style={Object {}}
>
            <TableRow
                        displayBorder={true}
                        hoverable={false}
                        hovered={false}
                        selectable={true}
                        striped={false}
            >
                        <TableRowColumn
                                    hoverable={false}
                        >
                                    <TextInput
                                                label="Sal치rio bruto CLT"
                                                onChange={undefined}
                                    />
                        </TableRowColumn>
                        <TableRowColumn
                                    hoverable={false}
                        >
                                    <TextInput
                                                label="Faturamento mensal MEI"
                                                onChange={undefined}
                                    />
                        </TableRowColumn>
            </TableRow>
</TableBody>,
          <TableFooter
            adjustForCheckbox={false}
            style={Object {}}
>
            <TableRow
                        displayBorder={true}
                        hoverable={false}
                        hovered={false}
                        selectable={true}
                        striped={false}
                        style={
                                    Object {
                                                "borderTop": "0px",
                                              }
                        }
            >
                        <TableRowColumn
                                    hoverable={false}
                                    style={
                                                Object {
                                                            "textAlign": "center",
                                                          }
                                    }
                        >
                                    <RaisedButton
                                                disabled={false}
                                                fullWidth={false}
                                                label="Calcular"
                                                labelPosition="after"
                                                onClick={undefined}
                                                primary={true}
                                                secondary={false}
                                    />
                        </TableRowColumn>
            </TableRow>
</TableFooter>,
        ],
        "fixedFooter": true,
        "fixedHeader": true,
        "height": "inherit",
        "id": "inputs",
        "multiSelectable": false,
        "selectable": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "adjustForCheckbox": false,
            "children": <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableHeaderColumn
                            style={
                                          Object {
                                                        "textAlign": "center",
                                                      }
                            }
              >
                            CLT
              </TableHeaderColumn>
              <TableHeaderColumn
                            style={
                                          Object {
                                                        "textAlign": "center",
                                                      }
                            }
              >
                            PJ
              </TableHeaderColumn>
</TableRow>,
            "displaySelectAll": false,
            "enableSelectAll": true,
            "selectAllSelected": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableHeaderColumn
                  style={
                                    Object {
                                                      "textAlign": "center",
                                                    }
                  }
>
                  CLT
</TableHeaderColumn>,
                <TableHeaderColumn
                  style={
                                    Object {
                                                      "textAlign": "center",
                                                    }
                  }
>
                  PJ
</TableHeaderColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "CLT",
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": "CLT",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "PJ",
                  "style": Object {
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": "PJ",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "allRowsSelected": false,
            "children": <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
>
              <TableRowColumn
                            hoverable={false}
              >
                            <TextInput
                                          label="Sal치rio bruto CLT"
                                          onChange={undefined}
                            />
              </TableRowColumn>
              <TableRowColumn
                            hoverable={false}
              >
                            <TextInput
                                          label="Faturamento mensal MEI"
                                          onChange={undefined}
                            />
              </TableRowColumn>
</TableRow>,
            "deselectOnClickaway": true,
            "displayRowCheckbox": false,
            "multiSelectable": false,
            "preScanRows": true,
            "selectable": true,
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TableRowColumn
                  hoverable={false}
>
                  <TextInput
                                    label="Sal치rio bruto CLT"
                                    onChange={undefined}
                  />
</TableRowColumn>,
                <TableRowColumn
                  hoverable={false}
>
                  <TextInput
                                    label="Faturamento mensal MEI"
                                    onChange={undefined}
                  />
</TableRowColumn>,
              ],
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <TextInput
                    label="Sal치rio bruto CLT"
                    onChange={undefined}
/>,
                  "hoverable": false,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Sal치rio bruto CLT",
                    "onChange": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <TextInput
                    label="Faturamento mensal MEI"
                    onChange={undefined}
/>,
                  "hoverable": false,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "label": "Faturamento mensal MEI",
                    "onChange": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "adjustForCheckbox": false,
            "children": <TableRow
              displayBorder={true}
              hoverable={false}
              hovered={false}
              selectable={true}
              striped={false}
              style={
                            Object {
                                          "borderTop": "0px",
                                        }
              }
>
              <TableRowColumn
                            hoverable={false}
                            style={
                                          Object {
                                                        "textAlign": "center",
                                                      }
                            }
              >
                            <RaisedButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Calcular"
                                          labelPosition="after"
                                          onClick={undefined}
                                          primary={true}
                                          secondary={false}
                            />
              </TableRowColumn>
</TableRow>,
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <TableRowColumn
                hoverable={false}
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
>
                <RaisedButton
                                disabled={false}
                                fullWidth={false}
                                label="Calcular"
                                labelPosition="after"
                                onClick={undefined}
                                primary={true}
                                secondary={false}
                />
</TableRowColumn>,
              "displayBorder": true,
              "hoverable": false,
              "hovered": false,
              "selectable": true,
              "striped": false,
              "style": Object {
                "borderTop": "0px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <RaisedButton
                  disabled={false}
                  fullWidth={false}
                  label="Calcular"
                  labelPosition="after"
                  onClick={undefined}
                  primary={true}
                  secondary={false}
/>,
                "hoverable": false,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "fullWidth": false,
                  "label": "Calcular",
                  "labelPosition": "after",
                  "onClick": undefined,
                  "primary": true,
                  "secondary": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
